### ISSUE TITLE
Pyright type errors

### ISSUE THREAD

**Issue by alexmojaki** [15r10nk/inline-snapshot#68]
We recently turned on type checking in tests and found that we had to add a bunch of `# type: ignore` comments to lines using `snapshot`. Here's a minimal repro:

```python
from inline_snapshot import snapshot

snapshot({})
snapshot([])
```

Output:

```
$ pyright file.py
file.py
  file.py:3:10 - error: Argument type is partially unknown
    Argument corresponds to parameter "obj" in function "snapshot"
    Argument type is "dict[Unknown, Unknown]" (reportUnknownArgumentType)
  file.py:4:10 - error: Argument type is partially unknown
    Argument corresponds to parameter "obj" in function "snapshot"
    Argument type is "list[Unknown]" (reportUnknownArgumentType)
2 errors, 0 warnings, 0 informations 
```

pyright wants to know the type of the argument, probably because of the `snapshot(obj: T) -> T` overload meaning it needs to know the type of `obj` to know what the return value is. And empty generic containers have unknown type arguments.

Maybe there's another way, but I think it's probably best to drop the overloads and just have `def snapshot(obj: Any=undefined) -> Any:`. It's not critical for users to typecheck the return value of `snapshot` since it will get asserted in tests anyway.

**Comment by 15r10nk** [comment]
Sorry, but I'm not able to reproduce your issue.

```shell
❯ bat file.py
───────┬────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: file.py
───────┼────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ from inline_snapshot import snapshot
   2   │ 
   3   │ 
   4   │ snapshot({})
   5   │ snapshot([])
   6   │ 
───────┴────────────────────────────────────────────────────────────────────────────────────────────────
 
❯ pyright --version
pyright 1.1.359

❯ pyright file.py
0 errors, 0 warnings, 0 informations 
```

**Comment by alexmojaki** [comment]
Try with this config:

```toml
[tool.pyright]
typeCheckingMode = 'strict'
```

**Comment by 15r10nk** [comment]
ok, I can reproduce your error, but I found out that `mypy --strict file.py` gives me 211 errors and complains that everything in inline-snapshot needs a type.

I don't know what the best approach here is. 
I will make a new release with this fix now.
I will later add types to inline-snapshot, which is probably a good idea in general.

**Comment by alexmojaki** [comment]
Thanks for fixing the issue.

I don't get any errors with `mypy --strict`. Maybe because I installed it from PyPI it didn't check the whole library, just the `snapshot` function?

**Comment by 15r10nk** [comment]
`mypy --strict` works also.

fixed in 0.8.1
